File: DuDoong-Domain/src/main/java/band/gosrock/domain/common/events/order/DoneOrderEvent.java
package band.gosrock.domain.common.events.order;


import band.gosrock.domain.common.aop.domainEvent.DomainEvent;
import band.gosrock.domain.domains.order.domain.Order;
import band.gosrock.domain.domains.order.domain.OrderMethod;
import lombok.Builder;
import lombok.Getter;
import lombok.ToString;
import org.springframework.lang.Nullable;

@Getter
@Builder
@ToString
public class DoneOrderEvent extends DomainEvent {

    private final String orderUuid;

    private final Long userId;

    private final OrderMethod orderMethod;

    @Nullable private final String paymentKey;

    private final Long itemId;

    public static DoneOrderEvent from(Order order) {
        return DoneOrderEvent.builder()
                .orderMethod(order.getOrderMethod())
                .paymentKey(order.isNeedPaid() ? order.getPaymentKey() : null)
                .userId(order.getUserId())
                .orderUuid(order.getUuid())
                .itemId(order.getItemId())
                .build();
    }
}

File: DuDoong-Domain/src/main/kotlin/band/gosrock/domain/common/events/order/DoneOrderEvent.kt
package band.gosrock.domain.common.events.order

import band.gosrock.domain.common.aop.domainEvent.DomainEvent
import band.gosrock.domain.domains.order.domain.Order
import band.gosrock.domain.domains.order.domain.OrderMethod

class DoneOrderEvent private constructor(
    val orderUuid: String,
    val userId: Long,
    val orderMethod: OrderMethod,
    val paymentKey: String?,
    val itemId: Long,
) : DomainEvent() {
    companion object {
        @JvmStatic
        fun from(order: Order): DoneOrderEvent = DoneOrderEvent(
            orderMethod = order.orderMethod!!,
            paymentKey = if (order.isNeedPaid()) order.paymentKey else null,
            userId = order.userId!!,
            orderUuid = order.uuid!!,
            itemId = order.itemId,
        )
    }
}
